Automattic/woocommerce-subscriptions-core
# Remove delayed change payment meta on new change payment requests

Related: https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3539#issuecomment-2437053113

## Description







While addressing Stripe deferred intent support for the Subscriptions Change Payment feature ([Stripe PR#3539](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3539)), we identified an issue where abandoned Change Payment requests would leave the '_delayed_update_payment_method_all' meta on the subscription. This causes subsequent Change Payment requests where the user has opted to update _all_ of their subscriptions to be ignored.

For instance, if you have 2 subscriptions with 4242 card and then initiate a request to change payment all subscriptions to Cash App, but never complete it, all future requests to update all subscriptions to another card `5555555555554444` will not update all of your subscriptions.

This PR fixes this issue by ensuring the '_delayed_update_payment_method_all' meta is deleted at the start of each new Change Payment request. This prevents stale meta from interfering with future requests.

## How to test this PR





1. Install the latest Stripe extension and have the new checkout enabled with cards and cash app payment methods available.
2. While on `trunk` of subscriptions-core, purchase two separate subscriptions using a standard 4242 card (if you don't already have multiple subscriptions)
3. Go to My Account > Subscriptions > View Subscription for one of your subscriptions.
4. Click the "Change payment" action and choose Cash App and make sure the "use this for all my subscriptions" checkbox is checked:![BM4Y7x.png](https://github.com/user-attachments/assets/a32dd1d0-e906-47c2-806b-da01a5b7e280)
5. Once the Cash App window opens, click the X to close the window:![1hwZWS.png](https://github.com/user-attachments/assets/951cab42-5e64-45dc-b78a-9883db9cc8e6)
6. Now try change to card `5555555555554444` and again, make sure the "use this for all my subscriptions" checkbox is checked
7. Go back to My Account > Subscriptions and notice that all of your subscriptions weren't updated: ![image](https://github.com/user-attachments/assets/5358b090-5477-4b56-815d-a5e44e73ec8e)
8. Checkout this branch and run the tests again, and confirm all subscriptions are properly updated to card `5555555555554444` after first attempting to update to Cash App

## Product impact
